app/Repositories/FileBaseRepository.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Repositories;
+
+class FileBaseRepository
+{
+    protected $dirName;
+    protected $fileName;
+
+    public function __construct($fileName, $dirName)
+    {
+        $this->fileName = $fileName;
+
+        if (!is_dir($dirName)) {
+            mkdir($dirName);
+        }
+        $this->filePath = prepare_file_path("{$dirName}/{$this->fileName}");
```

Comment:
please aslo define the filePath property in the class

```suggestion
        $this->filePath = prepare_file_path("{$dirName}/{$this->fileName}");

        if (!is_dir($dirName)) {
            mkdir($dirName);
        }

```